tests/OverrideTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Psalm\Tests;
+
+use Psalm\Config;
+use Psalm\Tests\Traits\InvalidCodeAnalysisTestTrait;
+use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
+
+use const DIRECTORY_SEPARATOR;
+
+class OverrideTest extends TestCase
+{
+    use ValidCodeAnalysisTestTrait;
+    use InvalidCodeAnalysisTestTrait;
+
+    protected function makeConfig(): Config
+    {
+        $config = parent::makeConfig();
+        $config->ensure_override_attribute = true;
+        return $config;
+    }
+
+    public function providerValidCodeParse(): iterable
+    {
+        return [
+            'constructor' => [
+                'code' => '<?php
+                    /**
+                     * @psalm-consistent-constructor
+                     */
+                    class C {
+                        public function __construct() {}
+                    }
+
+                    class C2 extends C {
+                        public function __construct() {}
+                    }
+                ',
+                'assertions' => [],
+                'ignored_issues' => [],
+                'php_version' => '8.3',
+            ],
+            'overrideClass' => [
+                'code' => '<?php
+                    class C {
+                        public function f(): void {}
+                    }
+
+                    class C2 extends C {
+                        #[Override]
+                        public function f(): void {}
+                    }
+                ',
+                'assertions' => [],
+                'ignored_issues' => [],
+                'php_version' => '8.3',
+            ],
+            'overrideInterface' => [
+                'code' => '<?php
+                    interface I {
+                        public function f(): void;
+                    }
+
+                    interface I2 extends I {
+                        #[Override]
+                        public function f(): void;
+                    }
+                ',
+                'assertions' => [],
+                'ignored_issues' => [],
+                'php_version' => '8.3',
+            ],
+        ];
+    }
+
+    public function providerInvalidCodeParse(): iterable
+    {
+        return [
+            'noParent' => [
+                'code' => '<?php
+                    class C {
+                        #[Override]
+                        public function f(): void {}
+                    }
+                ',
+                'error_message' => 'InvalidOverride - src' . DIRECTORY_SEPARATOR . 'somefile.php:3:25',
```

Comment:
The `error_message` is used for substring match, so you don't have to spell out the entire message here. 

```suggestion
                'error_message' => 'InvalidOverride',
```

Reply:
I saw some other tests doing this. I thought maybe it would be useful to keep the line and column number to ensure that the code location is correct. I don't feel strongly about it, though, and am fine with the change you've made.